sdks/git-sdk/src/main/java/org/jreleaser/sdk/git/ChangelogGenerator.java

```diff
@@ -251,12 +252,16 @@ private Version version(JReleaserContext context, Ref tag, Pattern versionPatter
         Pattern versionPattern = vp;
 
         unparseableTags.clear();
+
+        tags = tags.stream().filter(tagMatches(versionPattern)).collect(Collectors.toList());
```

Comment:
Something tells me this will break `early-access` releases as the tag that identifies such a release will be filtered before it can be checked.

How to test this:
- add tags v1.0.0 and v2.0.0
- add a commit after tag v2.0.0
- project version should be snapshot, such as 2.1.0-SNAPSHOT
- `gitService.getEffectiveTagName(context.getModel());` should return `early-access` in this case.
- generate changelog. Should have a single commit (from step 2)

Reply:
I'll do a test on that case.

Reply:
I think also this 'tags.removeAll(unparseableTags);' can break. What is the expected behavior here? If the project is a snapshot you are using 'early-access' as 'effectiveTagName' but why match all tags even if the tagName pattern is not matching?

Reply:
I want to add that this seems not working also before. The code inside the early-access branch

`tag = tags.stream()
                            .filter(ref -> !extractTagName(ref).equals(effectiveTagName))
                            .filter(ref -> currentVersion.equalsSpec(version(context, ref, versionPattern)))
                            .findFirst();`
Since in this case, the effectiveTagName is 'early-access' all tags will be filtered out.

Reply:
Correct. I think I have a fix for the last concern you raised, by inserting one additional filter

```
                        tag = tags.stream()
                            .filter(ref -> !extractTagName(ref).equals(effectiveTagName))
                            .filter(ref -> versionPattern.matcher(extractTagName(ref)).matches())
                            .filter(ref -> currentVersion.equalsSpec(version(context, ref, versionPattern)))
                            .findFirst();
```

Reply:
> I think also this 'tags.removeAll(unparseableTags);' can break. What is the expected behavior here? If the project is a snapshot you are using 'early-access' as 'effectiveTagName' but why match all tags even if the tagName pattern is not matching? So if I have a project in a snapshot mode I have to match again all possible tags not only the ones that matched the tagName pattern?

Correct. When a project is deemed as snapshot the `early-access` tag will compute the starting red which could be any of:

- an existing `early-access` tag
- the previously released tag, i.e. `v1.0.0`
- the start of the git history if no tags